src/tests/Interop/ICustomMarshaler/Primitives/ICustomMarshaler.cs

```diff
@@ -658,15 +677,15 @@ private class InnerCustomMarshaler : ILargeInterface, ICustomMarshaler
         [DllImport(LibcLibrary, EntryPoint = "atoi", CallingConvention = CallingConvention.Cdecl)]
         public static extern int DifferentCustomMarshalerType([MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(OuterCustomMarshaler))] string str);
 
-        [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(BoxedValueTypeCustomMarshaler))]
         public delegate string TestDelegateRef([MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(BoxedValueTypeCustomMarshaler))] ref int val);
 
         [DllImport("CustomMarshalersPrimitives", EntryPoint = "NativeParseIntDelegateRef")]
-        [return: MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(BoxedValueTypeCustomMarshaler))]
         public static extern string CustomMarshallerWithDelegateRef(int val, TestDelegateRef dlg);
 
         public static void DelegateParameter_MarshalerOnRefInt_ThrowsMarshalDirectiveException()
         {
+            // throw MarshalDirectiveException, but it is unhandled (?)
```

Comment:
We should probably fix this so that GetFunctionPointerForDelegate doesn't return "broken" function pointers that are just going to throw. Should be a separate pull request. Let's not enable this test for NativeAOT for now.

Reply:
Do this problem have tracking issue? I would like to change issue which prevent test from running to something different to indicate that https://github.com/dotnet/runtimelab/issues/160 indeed closed.

Reply:
Can I hide that test under 
```
if (RuntimeInformation.OSPlatform == Platform.Windows && !RuntimeFeatured.IsDynamicCodeSupported)
```

that way I will capture your words in the test. Maybe just 
```
if (!RuntimeFeatured.IsDynamicCodeSupported)
```

Reply:
I think I rushing again. I agree with Michal, `GetFunctionPointerForDelegate` should throw if somebody pass delegate which contains broken marshaling patterns. That would automatically fix current issue because throw happens before delegate was passed to native code.

My idea to create(if needed) separate issue, and re-assign test to that issue, because right now all related tests for  ICustomMarshaler passing. This problem in my understanding related to any delegates marshaling which contains unsupported configuration.

Reply:
It would be best to use `src\libraries\Common\tests\TestUtilities\System\PlatformDetection.cs` for this.

Reply:
> I agree with Michal, GetFunctionPointerForDelegate should throw if somebody pass delegate which contains broken marshaling patterns.

FWIW, it is not how CoreCLR works. CoreCLR will throw only once you actually try to call the delegate.

Reply:
I cannot add `src\libraries\Common\tests\TestUtilities\System\PlatformDetection.cs` alone because it complains about `IsWindows` properties. If I add `src\libraries\Common\tests\TestUtilities\System\PlatformDetection.Windows.cs` test build complain about `GenerateDllImportAttribute`. If I add reference to `TestUtilities.csproj` then `src\tests\build.sh` complain about not restored assets for the project.

I look for usages of this project in the tests, but seems to be it is used only in libraries tests. Please advice.

Reply:
If it is hard to use the full implementation from libraries currently, I would create a simplified local implementation of  PlatformDetection for runtime tests, with the subset of properties, and use that instead.

@jkoritzinsky Do you happen to have anything better in the plan for the runtime test consolidation?

Reply:
I think for now, making a copy either locally or in the `CoreCLRTestLibrary` assembly is the way to go. As we move tests over, we might decide to consolidate some of these utility libraries.